src/main.js

```diff
@@ -88,11 +88,15 @@ const updateRules = () => {
 
   const rules = flow(
     flatMap('settings'),
-    flatMap(({ path: settingsPath, scopedProperties }) =>
+    // In Atom prior to v1.32.0-beta0, the key for the nested setting properties was
```

Comment:
Am I understanding this correctly that you are going into the private properties of the settings to get the values? Atom has [an API for settings](https://atom.io/docs/api/v1.32.2/Config#instance-get) to avoid issues like this breakage that you are attempting to fix here.

If that's what's going on I would _highly_ recommend taking this opportunity to use the public API to prevent future breakages.

Reply:
Thanks a lot for your comment, @Arcanemagus! I'd be happy to replace this with something that relies on the official API. I was originally thinking of using the root scope descriptor to select the relevant settings but Atom's way of determining the root scope descriptor doesn't take all document properties into account that are relevant in this case (for example the public ID of the DOCTYPE declaration or the name of the root element, both of which don't necessarily appear on the first line). So I ended up checking the active packages, which isn't something a linter / autocomplete package would typically do. Do you see a way to improve this / prevent further breakage?

Reply:
Okay, dug into the code attempting to understand what is actually going on here. From what I can tell you are looking for any package that defines `validation.rules` scoped to the `.text.xml` scope.

You can get _all_ package's settings at that scope with:
```js
atom.config.get('', { scope: ['.text.xml'] });
```
From there you would just need to find that path, similar to how you are doing currently.

It look like you also use the path of the settings file to try to get the path of the package, you can get that from the API with [PackageManager::resolvePackagePath](https://atom.io/docs/api/v1.32.2/PackageManager#instance-resolvePackagePath).

I _think_ that would get all that you need using the API, but I've probably missed something 😛.

Reply:
that looks very promising -- thank you!